src/io/iter_image_recordio.cc

```diff
@@ -449,7 +453,20 @@ MXNET_REGISTER_IO_ITER(ImageRecordIter)
     return new PrefetcherIter(
         new BatchLoader(
             new ImageNormalizeIter(
-                new ImageRecordIter())));
+                new ImageRecordIter<real_t>())));
+  });
+
+MXNET_REGISTER_IO_ITER(ImageRecordUInt8Iter)
```

Comment:
Augmenter should work. The procedure is "rec -> augs -> InstVector<DType>". But DType normalize is not supported yet.